config/src/main/java/org/springframework/security/config/http/DefaultFilterChainValidator.java

```diff
@@ -75,11 +74,12 @@ private void checkPathOrder(List<SecurityFilterChain> filterChains) {
 		// Check that the universal pattern is listed at the end, if at all
 		Iterator<SecurityFilterChain> chains = filterChains.iterator();
 		while (chains.hasNext()) {
-			RequestMatcher matcher = ((DefaultSecurityFilterChain) chains.next()).getRequestMatcher();
-			if (AnyRequestMatcher.INSTANCE.equals(matcher) && chains.hasNext()) {
-				throw new IllegalArgumentException("A universal match pattern ('/**') is defined "
-						+ " before other patterns in the filter chain, causing them to be ignored. Please check the "
-						+ "ordering in your <security:http> namespace or FilterChainProxy bean configuration");
+			if (chains.next() instanceof DefaultSecurityFilterChain securityFilterChain) {
+				if (AnyRequestMatcher.INSTANCE.equals(securityFilterChain.getRequestMatcher()) && chains.hasNext()) {
+					throw new IllegalArgumentException("A universal match pattern ('/**') is defined "
```

Comment:
By adding this check, the one in `WebSecurity` is now redundant.  Will you please remove the related error message from `WebSecurity` and replace this one here with that one?

Partially, this is because this one refers to an XML block, which wouldn't apply in a Java Config application. Also, I want to preserve recent enhancements to the `WebSecurity` error message